### PR TITLE
Tolerate the case if `related` event cannot be found

### LINKED ISSUE #80146: events v1beta1: absence of optional `Related` event gives an error
Per the comments in v1beta1 Event:

https://github.com/kubernetes/kubernetes/blob/7285829b8306a7cc1a2dc0da943afcfe45c06579/staging/src/k8s.io/api/events/v1beta1/types.go#L61-L64

The `Related` field claimed to be optional, but an error is dumped if it's illegal - including the `nil` case:

https://github.com/kubernetes/kubernetes/blob/7285829b8306a7cc1a2dc0da943afcfe45c06579/staging/src/k8s.io/client-go/tools/events/event_recorder.go#L52-L55

This doesn't look right as it means the caller which passes in `nil` Related field will always get an err, for example, the scheduler binding event:

https://github.com/kubernetes/kubernetes/blob/7285829b8306a7cc1a2dc0da943afcfe45c06579/pkg/scheduler/scheduler.go#L440

If appropriate, I'm gonna raise a PR to tolerate the nil case. @yastij please advice.

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind bug
/sig scheduling
/sig api-machinery
/assign @yastij

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #80146

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/events/event_recorder.go [hunk 1]
```diff
@@ -49,9 +49,12 @@ func (recorder *recorderImpl) Eventf(regarding runtime.Object, related runtime.O
 		klog.Errorf("Could not construct reference to: '%#v' due to: '%v'. Will not report event: '%v' '%v' '%v'", regarding, err, eventtype, reason, message)
 		return
 	}
-	refRelated, err := reference.GetReference(recorder.scheme, related)
-	if err != nil {
-		klog.Errorf("Could not construct reference to: '%#v' due to: '%v'.", related, err)
+	var refRelated *v1.ObjectReference
+	if related != nil {
```
Comment (Huang-Wei) [inline staging/src/k8s.io/client-go/tools/events/event_recorder.go]: right now I'm making it just tolerate the `nil` case. Please let me know if you think we should tolerate all cases.

Comment (yastij) [reply staging/src/k8s.io/client-go/tools/events/event_recorder.go]: We should just switch to Infof as log level

**Review by yastij — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/client-go/tools/events/event_recorder.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/client-go/tools/events/event_recorder.go]: I would even say Infof.V(3) or sth...

Comment (yastij) [reply staging/src/k8s.io/client-go/tools/events/event_recorder.go]: thinking more about this, I'd put a `V(9)` as this would be used only for debug

Comment (Huang-Wei) [reply staging/src/k8s.io/client-go/tools/events/event_recorder.go]: I'm changing the logic to `V(9)`. Please take another look.

**Comment by Huang-Wei** [comment]
Thanks, @yastij @wojtek-t!

**Pull request merged**